Holovibes/sources/gui/windows/panels/export_panel.cc

```diff
@@ -45,34 +44,77 @@ void actualise_record_output_file_ui(const std::filesystem::path file_path)
 
 void ExportPanel::on_notify()
 {
-    if (api::get_compute_mode() == Computation::Raw)
+    // 0 = .holo, 1 = .mp4, 2 = .avi, 3 = .csv, 4 = .txt
+    static const std::map<RecordMode, std::vector<int>> extension_index_map = {{RecordMode::RAW, {0}},
+                                                                               {RecordMode::CHART, {3, 4}},
+                                                                               {RecordMode::HOLOGRAM, {0, 1, 2}},
+                                                                               {RecordMode::MOMENTS, {0}},
+                                                                               {RecordMode::CUTS_XZ, {1, 2}},
+                                                                               {RecordMode::CUTS_YZ, {1, 2}}};
```

Comment:
Not fan of these magic numbers. you can create a new enum like `OutputFormat`.
And I think this logic will be better in the API. I mean a function like `std::vector<OutputFormat> supported_formats(RecordMode mode);` that takes a mode and return an array of the supported output format. It makes no sense that this check is in the UI

Reply:
It's better but i think an enum in the API will be the best. It's cleaner but the Logic is in the UI while it should be in the API.

Reply:
I get it, I'll get to it!